### ISSUE TITLE
Tensorflow Training Crashing

### ISSUE THREAD

**Issue by nauyan** [tensorflow/tensorflow#50205]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): N/A
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): 2.2
- Python version: 3.8
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 11.3
- GPU model and memory: Tesla K80 and 12GB

I have created a GCP VM with Tesla K80 GPU attached to it. I have installed Nvidia 465 drivers for Ubuntu 20.04 along with Cuda 11.

I am trying to use tensorflow on the GCP machine and each time when the training starts the machine crashes after few epochs. Here is the log

```
216/216 [==============================] - ETA: 0s - loss: 2.5774 - accuracy: 0.2203   
216/216 [==============================] - 173s 800ms/step - loss: 2.5774 - accuracy: 0.2203 - val_loss: 47.4114 - val_accuracy: 0.1372 - lr: 0.0100
Epoch 2/50
216/216 [==============================] - ETA: 0s - loss: 1.9055 - accuracy: 0.3265  
216/216 [==============================] - 137s 633ms/step - loss: 1.9055 - accuracy: 0.3265 - val_loss: 46.8945 - val_accuracy: 0.2023 - lr: 0.0100
Epoch 3/50
216/216 [==============================] - ETA: 0s - loss: 1.7601 - accuracy: 0.3899  
216/216 [==============================] - 137s 633ms/step - loss: 1.7601 - accuracy: 0.3899 - val_loss: 1.9010 - val_accuracy: 0.3895 - lr: 0.0100
Epoch 4/50
216/216 [==============================] - ETA: 0s - loss: 1.5993 - accuracy: 0.4417  
216/216 [==============================] - 137s 632ms/step - loss: 1.5993 - accuracy: 0.4417 - val_loss: 1.7880 - val_accuracy: 0.3919 - lr: 0.0100
Epoch 5/50
216/216 [==============================] - ETA: 0s - loss: 1.2965 - accuracy: 0.5580  
216/216 [==============================] - 134s 618ms/step - loss: 1.2965 - accuracy: 0.5580 - val_loss: 1.9468 - val_accuracy: 0.3919 - lr: 0.0100
Epoch 6/50
 60/216 [=======>......................] - ETA: 1:20 - loss: 1.0874 - accuracy: 0.63542021-06-10 19:12:36.997237: E tensorflow/stream_executor/cuda/cuda_event.cc:29] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS: an illegal memory access was encountered
2021-06-10 19:12:36.997296: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:273] Unexpected Event status: 1
Aborted (core dumped)

```

Please advise if you have run into a similar sort of error before.

**Comment by saikumarchalla** [comment]
@nauyan  Please provide the simple standalone code to reproduce the issue at our end.Thanks

**Comment by nauyan** [comment]
```
from tensorflow.keras.preprocessing.image import ImageDataGenerator
import tensorflow
import matplotlib.pyplot as plt

# config = tensorflow.compat.v1.ConfigProto()
# config.gpu_options.allow_growth = True
# session = tensorflow.compat.v1.Session(config=config)


batch_size = 8
image_dir = "data/TrainingFrames/"
img_height, img_width = 256, 256
train_datagen = ImageDataGenerator(rescale=1./255,
    #shear_range=0.2,
    #zoom_range=0.2,
    #horizontal_flip=True,
    validation_split=0.2) # set validation split

train_generator = train_datagen.flow_from_directory(
    image_dir,
    target_size=(img_height, img_width),
    batch_size=batch_size,
    class_mode='categorical',
    subset='training') # set as training data

validation_generator = train_datagen.flow_from_directory(
    image_dir, # same directory as training data
    target_size=(img_height, img_width),
    batch_size=batch_size,
    class_mode='categorical',
    subset='validation') # set as validation data



from classification_models.tfkeras import Classifiers

ResNet18, preprocess_input = Classifiers.get('seresnet50')
model = ResNet18((img_height, img_width, 3), weights=None, classes=10)

optimizer = tensorflow.keras.optimizers.Adam(learning_rate=0.01) # 0.001

model.compile(optimizer=optimizer, loss = tensorflow.keras.losses.CategoricalCrossentropy(), metrics=['accuracy'])


EPOCHS = 50

# STEPS_PER_EPOCH = TRAINSET_SIZE // BATCH_SIZE
# VALIDATION_STEPS = VALSET_SIZE // BATCH_SIZE


callbacks = [
    tensorflow.keras.callbacks.ModelCheckpoint('results/weights/best.h5', monitor='val_accuracy', mode='max', verbose=1, save_best_only=True, save_weights_only=False),
    tensorflow.keras.callbacks.ReduceLROnPlateau(monitor='val_loss', factor=0.1, patience=8, min_lr=0.00001)
]




results = model.fit(train_generator, epochs=EPOCHS,
                        #steps_per_epoch=STEPS_PER_EPOCH,
                        #validation_steps=VALIDATION_STEPS,
                        callbacks=callbacks,
                        validation_data=validation_generator,
                        use_multiprocessing=False)

plt.figure(figsize=(8, 8))
plt.title("Learning curve")
plt.plot(results.history["loss"], label="loss")
plt.plot(results.history["val_loss"], label="val_loss")
plt.plot( np.argmin(results.history["val_loss"]), np.min(results.history["val_loss"]), marker="x", color="r", label="best model")
plt.xlabel("Epochs")
plt.ylabel("log_loss")
plt.legend();
plt.savefig('./results/plots/train_loss.png')



plt.figure(figsize=(8, 8))
plt.title("Learning curve")
plt.plot(results.history["accuracy"], label="accuracy")
plt.plot(results.history["val_accuracy"], label="val_accuracy")
plt.plot( np.argmax(results.history["val_accuracy"]), np.max(results.history["val_accuracy"]), marker="x", color="r", label="best model")
plt.xlabel("Epochs")
plt.ylabel("accuracy")
plt.legend();
plt.savefig('./esults/plots/train_accuracy.png')
```

**Comment by ymodak** [comment]
> E tensorflow/stream_executor/cuda/cuda_event.cc:29] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS: an illegal memory access was encountered

From the above stacktrace it looks like you have incompatible cuda/cudnn/tf versions installed on your system.
Note that recently released TF 2.5 binary supports cuda 11.2 and cudnn 8.1
You may quickly want to check your example with google colab where above configuration is pre installed for use.
This will help us to determine if changing your system config helps to fix the problem.
Thanks!
